chain/filecoin/account.go

```diff
@@ -0,0 +1,294 @@
+		methodNum = abi.MethodNum(payload[0])
+		payload = payload[1:]
+	}
+	return &Tx{
+		msg: types.Message{
+			Version:    types.MessageVersion,
+			From:       filfrom,
+			To:         filto,
+			Value:      big.Int{Int: value.Int()},
+			Nonce:      nonce.Int().Uint64(),
+			GasFeeCap:  big.Int{Int: txBuilder.gasFeeCap.Int()},
+			GasLimit:   txBuilder.gasLimit.Int().Int64(),
+			GasPremium: big.Int{Int: txBuilder.gasPremium.Int()},
+			Method:     methodNum,
+			Params:     payload,
+		},
+		signature: pack.Bytes65{},
+	}, nil
+}
+
+// ClientOptions are used to parameterise the behaviour of the Client.
+type ClientOptions struct {
+	MultiAddress string
+	AuthToken    string
+}
+
+// DefaultClientOptions returns ClientOptions with the default settings. These
+// settings are valid for use with the default local deployment of the
+// multichain. In production, the multi-address and authentication token should
+// be changed.
+func DefaultClientOptions() ClientOptions {
+	return ClientOptions{
+		MultiAddress: DefaultClientMultiAddress,
+		AuthToken:    DefaultClientAuthToken,
+	}
+}
+
+// WithAddress returns a modified version of the options with the given API
+// multi-address.
+func (opts ClientOptions) WithAddress(multiAddr string) ClientOptions {
+	opts.MultiAddress = multiAddr
+	return opts
+}
+
+// WithAuthToken returns a modified version of the options with the given
+// authentication token.
+func (opts ClientOptions) WithAuthToken(authToken string) ClientOptions {
+	opts.AuthToken = authToken
+	return opts
+}
+
+// Client holds options to connect to a filecoin lotus node, and the underlying
+// RPC client instance.
+type Client struct {
+	opts   ClientOptions
+	node   api.FullNode
+	closer jsonrpc.ClientCloser
+}
+
+// NewClient creates and returns a new JSON-RPC client to the Filecoin node
+func NewClient(opts ClientOptions) (*Client, error) {
+	requestHeaders := make(http.Header)
+	if opts.AuthToken != DefaultClientAuthToken {
+		requestHeaders.Add(AuthorizationKey, opts.AuthToken)
+	}
+
+	node, closer, err := filclient.NewFullNodeRPC(context.Background(), opts.MultiAddress, requestHeaders)
+	if err != nil {
+		return nil, err
+	}
+
+	return &Client{opts, node, closer}, nil
+}
+
+// Tx returns the transaction uniquely identified by the given transaction
+// hash. It also returns the number of confirmations for the transaction.
+func (client *Client) Tx(ctx context.Context, txID pack.Bytes) (account.Tx, pack.U64, error) {
+	// parse the transaction ID to a message ID
+	msgID, err := cid.Parse([]byte(txID))
+	if err != nil {
+		return nil, pack.NewU64(0), fmt.Errorf("parsing txID: %v", err)
+	}
+
+	// lookup message receipt to get its height
+	messageLookup, err := client.node.StateSearchMsg(ctx, msgID)
+	if err != nil {
+		return nil, pack.NewU64(0), fmt.Errorf("searching msg: %v", err)
+	}
+	if messageLookup == nil {
+		// FIXME: should this be an error? If the message is not yet found on-chain
+		return nil, pack.NewU64(0), nil
```

Comment:
An error would provide more semantics to the caller. Consider:

```suggestion
		return nil, pack.NewU64(0), fmt.Errorf("searching state for txid %v: not found", txID)
```